com.bktus.gpgfrontend.yaml

```
@@ -0,0 +1,96 @@
+app-id: com.bktus.gpgfrontend
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: GpgFrontend
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=host
+  - --device=dri
+  - --share=network
+
+cleanup:
+  - /lib/*.a
+  - /lib/*.la
+  - /lib32/*.a
+  - /lib32/*.la
+  - /share/doc
+  - /share/gtk-doc
+  - /share/man
+
+modules:
+  - name: gettext
+    buildsystem: simple
+    build-commands:
+      - ./configure --prefix=${FLATPAK_DEST} && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://ftp.gnu.org/pub/gnu/gettext/gettext-0.19.8.tar.gz
+        sha256: 3da4f6bd79685648ecf46dab51d66fcdddc156f41ed07e580a696a38ac61d48f
+  - name: libgpg-error
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://gnupg.org/ftp/gcrypt/libgpg-error/libgpg-error-1.42.tar.bz2
+        sha256: fc07e70f6c615f8c4f590a8e37a9b8dd2e2ca1e9408f8e60459c67452b925e23
+  - name: libassuan
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://gnupg.org/ftp/gcrypt/libassuan/libassuan-2.5.5.tar.bz2
+        sha256: 8e8c2fcc982f9ca67dcbb1d95e2dc746b1739a4668bc20b3a3c5be632edb34e4
+  - name: gpgme
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.20.0.tar.bz2
+        sha256: 25a5785a5da356689001440926b94e967d02e13c49eb7743e35ef0cf22e42750
+  - name: pinentry
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
```

```suggestion
    buildsystem: autotools
    config-opts:
      - --enable-maintainer-mode
```

Should use autotools, rest is figured out by flatpak-builder

---

Do you need the Qt or any other backends? What you did builds only the curses one. If so add `--enable-pinentry-qt`

---

Is `/usr/bin` hardcoded in your program?

```
[14:06:40.700] [T:24] [core] [ error  ] [GpgCoreInit.cpp:159] [init_gpgfrontend_core] -> core loaded custom gpgconf path is illegal: gpgconf (+2us)
[14:06:40.700] [T:24] [core] [warning ] [GlobalSettingStation.h:78] [LookupSettings] -> setting not found: network.auto_import_missing_key (+0us)
[14:06:40.700] [T:24] [core] [ error  ] [GpgCoreInit.cpp:172] [init_gpgfrontend_core] -> core loaded custom gpg key database is illegal:  (+35us)
QProcess::readAllStandardError: Called with MergedChannels
```

programs in the manifest goes into `/app`, pinentry is in `/app/bin` inside flatpak